### DISCUSSION TITLE
browse through timelaps mp4 video using mpv

### DISCUSSION THREAD

**Discussion by inftylux** [aaronwmorris/indi-allsky#1219] [Show and tell]
Certainly you all have your favorite way to watch your allsky mp4 videos, e.g to pick nice flares?
Here is my best practice (linux, others similar)

1. install command line viedeo viewer `mpv`

```
$ sudo apt install mpv 
```

2. define convenient key bindings, my favorite is using the mouse:

- WHEEL => back & forth one single frame (yes frame, not seconds!)
- Shift+WHEEL => back & forth five frames
- Alt+WHEEL => speed up/down
- Ctrl+WHEEL => zoom up/down
- Right-Click => toggle play/pause
- Left-Click => do screenshot
- Arrows => back & forth one single frame (yes frame, not seconds!)
here is my `~/.config/mpv/input.conf` to do so

```
WHEEL_UP frame-back-step
WHEEL_DOWN frame-step
Shift+WHEEL_UP frame-back-step;frame-back-step;frame-back-step;frame-back-step;frame-back-step
Shift+WHEEL_DOWN frame-step;frame-step;frame-step;frame-step;frame-step
Ctrl+WHEEL_UP add video-zoom 1
Ctrl+WHEEL_DOWN add video-zoom -1
Alt+WHEEL_UP add speed 0.1
Alt+WHEEL_DOWN add speed -0.1
LEFT frame-back-step
RIGHT frame-step
MBTN_LEFT screenshot
MBTN_RIGHT cycle pause

```

3. for convenience write a little bash wrapper `mpv.sh` with some specific options, e.g.
- start fullscreen
- start in pause 
- start with speed 0.1

```
#!/bin/bash
cat ~/.config/mpv/input.conf
mpv --fs --pause --speed=0.1  $1
```
the `cat` her is just for keeping in mind your choices...

4. arrange file association of mp4 with `mpv.sh`
 
5. browse your mp4
```
$ mpv.sh allsky-timelapse_ccd2_20240406_night.mp4
```

adapt and enjoy!

**Comment by aaronwmorris** [comment]
Watching videos could certainly be better in the browser.  This is definitely interesting.

**Reply by inftylux** [reply]
... that's just fine.  I'm being annoyed by positioning the videoslider to pick a single specific frame, just saw the flare flashing but never finding it again by manual rewinding... maybe you know some shortcuts for that?

**Reply by aaronwmorris** [reply]
I have the same troubles.  I usually try to pause the video just after the event, go to the Image Viewer at that timestamp and work my way backwards using the right arrow key.

You can also use the Gallery view to look at the hour and hopefully there is enough detail to show in the thumbnail so you can see it among the other images.
